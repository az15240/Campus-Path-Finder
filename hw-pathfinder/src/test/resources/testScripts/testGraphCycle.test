# Test a graph with cycles.
# Graph ('A') has three nodes being in a cycle
# Graph ('B') has three nodes being in a cycle and another node connecting to the cycle.
# Graph ('C') has two cycles of three nodes with a shared arc of two nodes.

CreateGraph A
AddNode A n1
AddNode A n2
AddNode A n3
# forms a cycle of (n1,n2,n3)
AddEdge A n1 n2 1.2
AddEdge A n2 n3 2.3
AddEdge A n3 n1 3.1
ListNodes A
ListChildren A n1
ListChildren A n2
ListChildren A n3
FindPath A n1 n3
FindPath A n3 n1

CreateGraph B
AddNode B n1
AddNode B n2
AddNode B n3
AddNode B n4
# forms a cycle of (n1,n2,n3)
AddEdge B n1 n2 1.2
AddEdge B n2 n3 2.3
AddEdge B n3 n1 3.1
# connect n4 to the cycle
AddEdge B n1 n4 1.4
AddEdge B n4 n1 4.1
ListNodes B
ListChildren B n1
ListChildren B n2
ListChildren B n3
ListChildren B n4
FindPath B n2 n4
FindPath B n4 n2

CreateGraph C
AddNode C n1
AddNode C n2
AddNode C n3
AddNode C n4
# forms a cycle of (n1,n2,n3)
AddEdge C n1 n2 1.2
AddEdge C n2 n3 2.3
AddEdge C n3 n1 3.1
# forms a cycle of (n1,n2,n4)
AddEdge C n1 n4 1.4
AddEdge C n4 n2 4.2
AddEdge C n2 n1 2.1
ListNodes C
ListChildren C n1
ListChildren C n2
ListChildren C n3
ListChildren C n4
FindPath C n3 n4
FindPath C n4 n3
