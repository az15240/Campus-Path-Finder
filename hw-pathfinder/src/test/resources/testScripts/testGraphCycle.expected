# Test a graph with cycles.
# Graph ('A') has three nodes being in a cycle
# Graph ('B') has three nodes being in a cycle and another node connecting to the cycle.
# Graph ('C') has two cycles of three nodes with a shared arc of two nodes.

created graph A
added node n1 to A
added node n2 to A
added node n3 to A
# forms a cycle of (n1,n2,n3)
added edge 1.200 from n1 to n2 in A
added edge 2.300 from n2 to n3 in A
added edge 3.100 from n3 to n1 in A
A contains: n1 n2 n3
the children of n1 in A are: n2(1.200)
the children of n2 in A are: n3(2.300)
the children of n3 in A are: n1(3.100)
path from n1 to n3:
n1 to n2 with weight 1.200
n2 to n3 with weight 2.300
total cost: 3.500
path from n3 to n1:
n3 to n1 with weight 3.100
total cost: 3.100

created graph B
added node n1 to B
added node n2 to B
added node n3 to B
added node n4 to B
# forms a cycle of (n1,n2,n3)
added edge 1.200 from n1 to n2 in B
added edge 2.300 from n2 to n3 in B
added edge 3.100 from n3 to n1 in B
# connect n4 to the cycle
added edge 1.400 from n1 to n4 in B
added edge 4.100 from n4 to n1 in B
B contains: n1 n2 n3 n4
the children of n1 in B are: n2(1.200) n4(1.400)
the children of n2 in B are: n3(2.300)
the children of n3 in B are: n1(3.100)
the children of n4 in B are: n1(4.100)
path from n2 to n4:
n2 to n3 with weight 2.300
n3 to n1 with weight 3.100
n1 to n4 with weight 1.400
total cost: 6.800
path from n4 to n2:
n4 to n1 with weight 4.100
n1 to n2 with weight 1.200
total cost: 5.300

created graph C
added node n1 to C
added node n2 to C
added node n3 to C
added node n4 to C
# forms a cycle of (n1,n2,n3)
added edge 1.200 from n1 to n2 in C
added edge 2.300 from n2 to n3 in C
added edge 3.100 from n3 to n1 in C
# forms a cycle of (n1,n2,n4)
added edge 1.400 from n1 to n4 in C
added edge 4.200 from n4 to n2 in C
added edge 2.100 from n2 to n1 in C
C contains: n1 n2 n3 n4
the children of n1 in C are: n2(1.200) n4(1.400)
the children of n2 in C are: n1(2.100) n3(2.300)
the children of n3 in C are: n1(3.100)
the children of n4 in C are: n2(4.200)
path from n3 to n4:
n3 to n1 with weight 3.100
n1 to n4 with weight 1.400
total cost: 4.500
path from n4 to n3:
n4 to n2 with weight 4.200
n2 to n3 with weight 2.300
total cost: 6.500